sklearn/utils/class_weight.py

```diff
@@ -51,9 +51,15 @@ def compute_class_weight(class_weight, classes, y):
         if not all(np.in1d(classes, le.classes_)):
             raise ValueError("classes should have valid labels that are in y")
 
-        recip_freq = len(y) / (len(le.classes_) *
-                               np.bincount(y_ind).astype(np.float64))
+        freq = np.bincount(y_ind).astype(np.float64)
+        recip_freq = len(y) / (len(le.classes_) * freq)
         weight = recip_freq[le.transform(classes)]
+        if np.any(np.diff(freq)):
```

Comment:
I don't understand this line. Doesn't that rely on the class order?

Reply:
Sorry, I had intended frequent*weight, to check if there was exact float equality under the weighting. But we could use this hack in all cases if we think that's better (less architecture-dependent for instance)

Reply:
I don't understand the hack then ;)